compiler/lib/src/main/scala/util/Error.scala

```diff
@@ -211,6 +215,10 @@ sealed trait Error {
         Error.print (Some(loc)) ("passive component may not have async input")
       case SemanticError.PassiveStateMachine(loc) =>
         Error.print (Some(loc)) ("passive component may not have a state machine instance")
+      case SemanticError.PortNumberAlreadyInUse(n) =>
+        Error.print (None) (s"port number $n is already in use")
+      case SemanticError.PortNumberOutOfRange(portNumber, port, arraySize) =>
```

Comment:
`PortNumberOutOfRange` appears to be redundant with `InvalidPortNumber`. Can we just use `InvalidPortNumber`?

Reply:
Update: we want a new error along the lines of "matched port numbering could not find an available port number on both sides of the matching"